src/middlewared/middlewared/plugins/pool.py

```diff
@@ -208,6 +208,14 @@ class PoolService(CRUDService):
         options['prefix'] = 'vol_'
         return await self.middleware.call('datastore.query', 'storage.volume', filters, options)
 
+    @accepts()
+    async def file_system_choices(self):
+        vol_names = [vol['name'] for vol in (await self.query())]
+        return [
+            y['name'] for y in await self.middleware.call('zfs.dataset.query', [('name', 'rnin', '.system')])
+            if y['name'].split('/')[0] in vol_names
```

Comment:
That for/if is not necessary, you can use a filter as well I suppose: `('pool', 'in', vol_names)`